src/ins-soap-client.service.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  Client,
+  ClientSSLSecurityPFX,
+  createClientAsync
+} from 'soap';
+import fs from 'fs';
+import moment from 'moment';
+import { v4 as uuidv4 } from 'uuid';
+import { AxiosResponse } from 'axios';
+import { combineCACertAsPem, PASSPHRASE } from './certificates';
+import { INSSoapClientHelper } from './ins-soap-client.helper';
+
+export interface SOAPINSConfig {
+  softwareName: string;
+  softwareVersion: string;
+  emitter: string;
+  IDAMAutorisationNumber: string;
+}
+
+export interface INSIdentityTraits {
+  firstName: string;
+  lastName: string;
+  gender: 'M' | 'F';
+  birthdate: Date,
+}
+
+export interface INSSearchFromIdentityTraits {
+  messageUuid: string;
+  result: AxiosResponse;
+}
+
+const SoapActions = {
+  ['searchFromIdentityTraits']: {
+    header: { Action: 'urn:ServiceIdentiteCertifiee:1.0.0:rechercherInsAvecTraitsIdentite' },
+    method: 'rechercherInsAvecTraitsIdentite',
+  },
+}
+
+export class INSSoapClientService {
+  private readonly _wsdlUrl: string = 'src/fixtures/WSDL/DESIR_ICIR_EXP_1.5.0.wsdl';
+
+  private readonly _config: SOAPINSConfig;
+
+  private readonly _bamCtxId: string;
+  private readonly _lpsCtxId: string;
+  private readonly _lpsCtxInstance: string;
+
+  private _soapClient: Client;
+
+  constructor(config: SOAPINSConfig) {
+    this._config = config;
+    this._bamCtxId = uuidv4();
```

Comment:
- How often do we need them to cycle?
- Does the lpsCtxId reflects the medical member's id in some way?
- With the instance being the employee id?

Reply:
how would I know ? My job was just to implement SOAP client, not understand all the fields of the web service ^^